### ISSUE TITLE
Undeclared dependencies leaks into unused-dependency

### ISSUE THREAD

**Issue by tobias-** [nebula-plugins/gradle-lint-plugin#181]
https://github.com/nebula-plugins/gradle-lint-plugin/blob/72c36b5e553fff19be7361d9e2cddc40da221c2e/src/main/groovy/com/netflix/nebula/lint/rule/dependency/UnusedDependencyRule.groovy#L109

```
error     unused-dependency                  one or more classes in com.fasterxml.jackson.core:jackson-annotations:2.9.4 are required by your code directly
```

Since the `UndeclaredDependencyRule.groovy` already takes care of this, the block should probably be deleted.

**Comment by xenomachina** [comment]
It would be *really* nice if this could get fixed. I'd like to use unused-dependency, but we get so many false positives due to "one or more classes in ... are required by your code directly" that it ends up being too noisy to use in practice. Many libraries *intentionally* have artifacts you can depend on to transitively pull in other artifacts. Also, since Gradle added the [api vs implementation separation](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation), it's far less likely that transitive dependencies will leak in unintentionally.

**Comment by tobias-** [comment]
I'm not using this anymore
